app/frontend/src/components/Clients.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <v-card>
+    <template>
+      <v-card-title>
+        Registered Service Clients
+        <v-spacer></v-spacer>
+        <v-text-field
+          v-model="search"
+          append-icon="mdi-magnify"
+          label="Search"
+          single-line
+          hide-details
+        ></v-text-field>
+      </v-card-title>
+
+      <v-data-table
+        class="kc-table"
+        dense
+        :headers="headers"
+        :items="serviceClients"
+        :items-per-page="5"
+        :search="search"
+        :loading="loading"
+        loading-text="Loading... Please wait"
+      >
+        <template v-slot:item.dev="{item}">
+          <v-icon v-if="item.dev === 1" color="green">check</v-icon>
+        </template>
+
+        <template v-slot:item.test="{item}">
+          <v-icon v-if="item.test === 1" color="green">check</v-icon>
+        </template>
+
+        <template v-slot:item.prod="{item}">
+          <v-icon v-if="item.prod === 1" color="green">check</v-icon>
+        </template>
+      </v-data-table>
+    </template>
+  </v-card>
+</template>
+
+
+<script>
+import keycloakService from '@/services/keycloakService';
+
+export default {
+  name: 'Clients',
+  data() {
+    return {
+      // vuetify data table
+      search: '',
+      headers: [
+        { text: 'Name', align: 'start', value: 'name' },
+        { text: 'DEV', value: 'dev' },
+        { text: 'TEST', value: 'test' },
+        { text: 'PROD', value: 'prod' }
+      ],
+      serviceClients: [],
+      loading: true
+    };
+  },
+  watch: {
+    // hide data table progress bar when serviceClients have been returned from backend
+    serviceClients() {
+      this.loading = false;
+    }
+  },
+  methods: {
+    // get table data from frontend service layer
+    getData() {
+      keycloakService.getServiceClients().then(response => {
+        if (response) {
+
+          // reformat serviceClients data into array of objects that will work for the data table
```

Comment:
I spent way too long trying to reformat the data using other array functions. this approach might be a bit crude but it's the only way i could get it working